File: cosmoauthorization/src/java/it/csi/cosmo/cosmoauthorization/business/service/ImpostazioniFirmaService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package it.csi.cosmo.cosmoauthorization.business.service;

import it.csi.cosmo.cosmoauthorization.dto.rest.ImpostazioniFirma;

/**
 * Servizio per la gestione delle impostazioni di firma
 */
public interface ImpostazioniFirmaService {

  /**
   * Restituisce le impostazioni di firma
   *
   * @return le impostazioni di firma
   */
  ImpostazioniFirma getImpostazioniFirma();

}

File: cosmowcl/src/app/shared/services/impostazioni-firma.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ImpostazioniFirma } from '../models/api/cosmoauthorization/impostazioniFirma';
import { ApiUrls } from '../utilities/apiurls';

@Injectable()
export class ImpostazioniFirmaService {

  constructor(
    private http: HttpClient) {
  }

  /** GET: Restituisce un le impostazioni di firma. */
  getImpostazioniFirma(): Observable<ImpostazioniFirma> {
    return this.http.get<ImpostazioniFirma>(ApiUrls.IMPOSTAZIONI_FIRMA);
  }
}
